### ISSUE TITLE
Clarification to your assumptions.

### ISSUE THREAD

**Issue by Mr-Nobody-dey** [somanchiu/ReSwapper#15]
> To resolve the smoothness issue on the face, I have an idea. I assume that the ArcFace model embedding can represent the face texture. We can then train a model using the same architecture as InSwapper and using supervised learning from the main branch, but train with skin-smoothed Image A (target) + Image A embedding (source) = Image A (ground truth). As a result, the model should be able to add any face texture back to the face.
> 
> As an extension of this approach, maybe we just need to add another "data augmentation" to train.py? 

 _Originally posted by @somanchiu in [#12](https://github.com/somanchiu/ReSwapper/issues/12#issuecomment-2588847747)_

ArcFace model embeddes the input image to a latent space of 512 shaped vector. So it Does not represent face texture. Although their exists few works where human face texture models exists. But I have no idea how good they are.

**Comment by blend-er** [comment]
the training of arcface doesn't explicitly exclude facial textures from being included in that 512-dim vector. that's why there's also an issue with pose being entangled in that vector as well. more importantly, the arcface model uses 112x112 input which may be too low-res to sufficiently capture any face texture at all.

we may need some pre-training scheme with only the encoder/down and decoder/up layers such that they learn a 1:1 identity function. if we examine inswapper_128 on FFHQ, we can see that they have a perfect reconstruction of the background with only changes to the relevant parts of the face.

**Comment by Mr-Nobody-dey** [comment]
yes. DeepFaceLab does extremely well in that respect, but it is too expensive to train.
